subsys/zigbee/osif/zb_nrf_transceiver.c

```diff
@@ -228,26 +283,32 @@ zb_bool_t zb_trans_is_active(void)
 zb_bool_t zb_trans_transmit(zb_uint8_t wait_type, zb_time_t tx_at,
 			    zb_uint8_t *tx_buf, zb_uint8_t current_channel)
 {
+	struct net_buf frag = {
+		.frags = NULL,
+		.b = { .data = &tx_buf[1],
+		       .len = tx_buf[0] - FCS_LENGTH,
+		       .size = tx_buf[0] - FCS_LENGTH,
+		       .__buf = &tx_buf[1]
+		}
+	};
+	int err = 0;
+
 	LOG_DBG("Function: %s, channel: %d", __func__, current_channel);
-	zb_bool_t transmit_status = ZB_FALSE;
 
 #ifndef ZB_ENABLE_ZGP_DIRECT
 	ARG_UNUSED(tx_at);
 	ARG_UNUSED(current_channel);
 #endif
 
+	ack_frame = NULL;
+
 	switch (wait_type) {
 	case ZB_MAC_TX_WAIT_CSMACA:
-#if !NRF_802154_CSMA_CA_ENABLED
-		while (!transmit_status) {
-			transmit_status = (zb_bool_t)nrf_802154_transmit_raw(
-				tx_buf, ZB_FALSE);
-		}
-		break;
-#else
-		transmit_status = ZB_TRUE;
-		nrf_802154_transmit_csma_ca_raw(tx_buf);
-#endif
+		state_cache.radio_state = RADIO_802154_STATE_TRANSMIT;
+		err = radio_api->tx(radio_dev,
+			IEEE802154_TX_MODE_CSMA_CA,
+			NULL,
```

Comment:
Have you verified that it's ok to pass a NULL `net_pkt`?

Reply:
Yes, `pkt` is not used inside tx function. The packet that will be sent is provided via `frag` parameter. 
Radio API header comments about tx function kind of silimiar way too:
`/** Transmit a packet fragment */`